docs/en/ingest-management/agent-policies.asciidoc

```diff
@@ -282,9 +282,9 @@ will still forward data to the cluster.
 
 When you create an integration policy you often need to provide sensitive information such as an API key or a password. To help ensure that data can't be accessed inappropriately, any secret values used in an integration policy are stored separately from other policy details.
 
-As well, after you've saved a secret value in {fleet}, the value is hidden in both the {fleet} UI and in the agent policy definition. When you view the agent policy (**Actions -> View policy**), an environment variable is displayed in place of any secret values, for example `${SECRET_0}`.
+When you add a secret value directly to the {kibana-ref}/fleet-settings-kb.html[`kibana.yml` configuration file], the value is replaced with a reference, with the original value stored outside of {fleet} and {kib} as a secure hash. Note that this type of secret storage requires all configured {fleet-server}s to be on version 8.12.0 or higher.
```

Comment:
I'm wondering whether this is the right location to add information about the `kibana.yml` config file. Right now, secret values in the `kibana.yml` file can only pertain to preconfigured outputs, all other secret values (in output or integration policies) are inputted via the UI. Since this section is under "Fleet UI settings", it might be a little confusing.

It doesn't look like there is an dedicated page for preconfigured outputs under "Manage Elastic Agents in Fleet", there are documented [here](https://www.elastic.co/guide/en/kibana/current/fleet-settings-kb.html). I'm thinking it would be good to add documentation to that page. The idea is that now a preconfigured output can specify secret values, for example in the case of a Logstash output, instead of:
```yml
xpack.fleet.outputs:
  - id: my-logstash-output
    name: preconfigured logstash output
    type:  logstash
    hosts: ["localhost:9999"]
    ssl:
      certificate: xxxxxxxxxx
      key: unsecurekey
```
We can have:
```yml
xpack.fleet.outputs:
  - id: my-logstash-output-with-a-secret
    name: preconfigured logstash output with a secret
    type:  logstash
    hosts: ["localhost:9999"]
    ssl:
      certificate: xxxxxxxxxx
    secrets:
      ssl:
        key: securekey
```

What do you think?

Reply:
Ah! I see. I misunderstood things. I agree that the [Fleet Settings page](https://www.elastic.co/guide/en/kibana/current/fleet-settings-kb.html) in the Kibana docs is a better home for this. I've opened https://github.com/elastic/kibana/pull/172637

Comparing that page with your example above, I noticed that the SSL settings aren't documented, so I've added them in. I also noticed that a couple of other settings are listed as required, but since they're not in your example I'm thinking they should be documented as optional. So I included that fix in the PR as well.